src/view/popup/popup.tsx

```diff
@@ -53,8 +53,9 @@ export const Popup = () => {
             {page === "landing" ? (
                 <LandingPage
                     onRecordButtonPress={async () => {
-                        goToPage("recording");
-                        await refreshAndGetSize();
+                        if (await refreshAndGetSize()) {
```

Comment:
can you reassign to a var instead of doing all of this inline ? like size = await refreshAndGetSize() then do the conditional using the var

Reply:
also having 'and' in a function name should be a code wuality smell as function should ideally just do one thing - this concept is mentioned quite heavily in books like 'clean code' and 'refactoring'